test/e2e/scheduler_predicates.go

```diff
@@ -1280,4 +1281,234 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
+			TypeMeta: unversioned.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: podName,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  podName,
+						Image: "gcr.io/google_containers/pause-amd64:3.0",
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(framework.WaitForPodRunningInNamespace(c, podName, ns))
+		pod, err := c.Pods(ns).Get(podName)
+		framework.ExpectNoError(err)
+
+		nodeName := pod.Spec.NodeName
+		err = c.Pods(ns).Delete(podName, api.NewDeleteOptions(0))
+		framework.ExpectNoError(err)
+
+		By("Trying to apply a random taint on the found node.")
+		taintName := fmt.Sprintf("kubernetes.io/e2e-taint-key-%s", string(util.NewUUID()))
+		taintValue := "testing-taint-value"
+		taintEffect := string(api.TaintEffectNoSchedule)
+		framework.RunKubectlOrDie("taint", "nodes", nodeName, taintName+"="+taintValue+":"+taintEffect)
+		By("verifying the node has the taint " + taintName + " with the value " + taintValue)
+		output := framework.RunKubectlOrDie("describe", "node", nodeName)
+		requiredStrings := [][]string{
+			{"Name:", nodeName},
+			{"Taints:"},
+			{taintName, taintValue, taintEffect},
+		}
+		checkOutput(output, requiredStrings)
+
+		By("Trying to apply a random label on the found node.")
+		labelKey := fmt.Sprintf("kubernetes.io/e2e-label-key-%s", string(util.NewUUID()))
+		labelValue := "testing-label-value"
+		framework.RunKubectlOrDie("label", "nodes", nodeName, labelKey+"="+labelValue)
+		By("verifying the node has the label " + labelKey + " with the value " + labelValue)
+		labelOutput := framework.RunKubectlOrDie("describe", "node", nodeName)
+		labelOutputRequiredStrings := [][]string{
+			{"Name:", nodeName},
+			{"Labels:"},
+			{labelKey + "=" + labelValue},
+		}
+		checkOutput(labelOutput, labelOutputRequiredStrings)
+
+		By("Trying to the pod, still no tolerations.")
+		podNameNoTolerations := "still-no-tolerations"
+		_, err = c.Pods(ns).Create(&api.Pod{
+			TypeMeta: unversioned.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: podNameNoTolerations,
+			},
+			Spec: api.PodSpec{
+				NodeSelector: map[string]string{labelKey: labelValue},
+				Containers: []api.Container{
+					{
+						Name:  podNameNoTolerations,
+						Image: "gcr.io/google_containers/pause-amd64:3.0",
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err)
+		// Wait a bit to allow scheduler to do its thing
+		// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
+		framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
+		time.Sleep(10 * time.Second)
+
+		verifyResult(c, podNameNoTolerations, ns)
+		cleanupPods(c, ns)
+
+		By("removing the taint " + taintName + " off the node " + nodeName)
+		framework.RunKubectlOrDie("taint", "nodes", nodeName, taintName+"-")
+		By("verifying the node doesn't have the taint " + taintName)
```

Comment:
Add one more step, where you verify that the pod gets schedule after you remove the taint.